sklearn/covariance/robust_covariance.py

```diff
@@ -435,7 +434,6 @@ def fast_mcd(X, support_fraction=None,
             # (and less optimal)
             all_best_covariances = np.zeros((n_best_tot, n_features,
                                              n_features))
-            n_best_tot = 10
```

Comment:
Doesn't this look odd? The comment says it's trying a smaller matrix, but it sets the `n_best_tot` after retrying the matrix allocation. It seems to me that those lines are swapped somehow.

Reply:
Good point. I'll try swapping the lines and see if anything breaks.